lib/trace.js

```diff
@@ -461,14 +461,16 @@ Trace.prototype.getLoadRecord = function(canonical, excludeURLs, parentStack) {
             if (typeof sourceMap == 'string')
               sourceMap = load.metadata.sourceMap = JSON.parse(sourceMap);
 
-            var originalName = load.name.split('!')[0];
-            
-            // force set the filename of the original file
-            sourceMap.file = originalName + '!transpiled';
-
-            // force set the sources list if only one source
-            if (!sourceMap.sources || sourceMap.sources.length <= 1)
-              sourceMap.sources = [originalName];
+            if (load.name.indexOf('!') != -1) {
```

Comment:
Ok, we should just be able to replace this with `if (load.metadata.loader)` rather than plugin syntax specifically, so it applies to all plugins.

I will then also make this change in https://github.com/systemjs/systemjs/blob/master/lib/plugins.js#L202 for the next SystemJS release to match.

Let me know if that works for you?